addrmgr/addrmanager.go

```diff
@@ -170,6 +170,8 @@ func (a *AddrManager) updateAddress(netAddr, srcAddr *wire.NetAddress) {
 	addr := NetAddressKey(netAddr)
 	ka := a.find(netAddr)
 	if ka != nil {
+		ka.mtx.Lock()
+		defer ka.mtx.Unlock()
```

Comment:
Defer operates at the function scope.  This will not release the mutex at the end of the if statement scope as I suspect you intended.